src/uzbl-ext.c

```diff
@@ -167,3 +189,60 @@ dom_blur_callback (WebKitDOMEventTarget *target,
     uzbl_extio_send_new_message (g_io_stream_get_output_stream (ext->stream),
                                  EXT_BLUR, name);
 }
+
+void
+ext_scroll_web_page (UzblExt      *ext,
+                     ScrollAxis    axis,
+                     ScrollAction  action,
+                     ScrollMode    mode,
+                     gint32       amount)
+{
+    WebKitDOMDocument *doc = webkit_web_page_get_dom_document (ext->web_page);
+    WebKitDOMDOMWindow *win = webkit_dom_document_get_default_view (doc);
+    WebKitDOMElement *body = WEBKIT_DOM_ELEMENT (
+        webkit_dom_document_get_body (doc));
+
+    if (action == SCROLL_BEGIN) {
+        webkit_dom_dom_window_scroll_to (
+            win,
+            axis == SCROLL_HORIZONTAL ? 0
+                                      : webkit_dom_dom_window_get_scroll_x (win),
+            axis == SCROLL_HORIZONTAL ? webkit_dom_dom_window_get_scroll_y (win)
+                                      : 0
+        );
+    } else if (action == SCROLL_END) {
+        webkit_dom_dom_window_scroll_to (
+            win,
+            axis == SCROLL_HORIZONTAL ? webkit_dom_element_get_scroll_width (body)
+                                      : webkit_dom_dom_window_get_scroll_x (win),
+            axis == SCROLL_HORIZONTAL ? webkit_dom_dom_window_get_scroll_y (win)
+                                      : webkit_dom_element_get_scroll_height (body)
+        );
+    } else if (action == SCROLL_SET) {
+        if (mode == SCROLL_PERCENTAGE) {
+            glong page = (axis == SCROLL_HORIZONTAL) ? webkit_dom_element_get_scroll_width (body)
+                                                     : webkit_dom_element_get_scroll_height (body);
+            amount = (gint32) page * amount * 0.01;
```

Comment:
the old code had a concept lower and upper values that was queried from gtk in dom land nothing like this exist. It might make sense to deduct the view port size here as scrolling to 100% of the page is nonsense request. Scrolling is defined by where the top of the viewport and that would put it entirely outside of the page.